packages/vite/src/node/server/index.ts

```diff
@@ -1154,6 +1154,17 @@ export function resolveServerOptions(
     )
   }
 
+  if (
+    process.env.__VITE_ADDITIONAL_SERVER_ALLOWED_HOSTS &&
+    Array.isArray(server.allowedHosts)
+  ) {
+    const additionalHosts = process.env.__VITE_ADDITIONAL_SERVER_ALLOWED_HOSTS
+      .split(',')
+      .map((h) => h.trim())
+
+    server.allowedHosts = [...server.allowedHosts, ...additionalHosts]
```

Comment:
For now, we want to limit the env var to only allow a single additional host (if that's fine for your usecase).
```suggestion
    const additionalHost = process.env.__VITE_ADDITIONAL_SERVER_ALLOWED_HOSTS.trim()
    server.allowedHosts = [...server.allowedHosts, additionalHost]
```